script.js

```diff
@@ -0,0 +1,134 @@
+
+let paginatedBtnGroup= document.querySelector(".paginated-btn-grp");
+// let currentPageNumber=1;
+let noOfPosts=6;
+let totalPage =0;
+let dataList=[];
+let currentPage=1;
+
+async function getAllData(){
+    
+    const data=await fetch('https://dummyjson.com/posts');
+    const json=await data.json();
+    //  dataList= json.posts;
+      dataList=json.posts;
+     console.log(dataList);
+
+     totalPage=Math.ceil(dataList.length / noOfPosts);
+     
+     createPaginatedButton(totalPage);
+     displayItems(dataList,currentPage); 
+
+    
+}
+
+async function fetchDataAndUseData() {
+    await getAllData(currentPage);
+    console.log(dataList); // Now dataList will be populated
+    // You can use dataList here or pass it to another function
+}
+fetchDataAndUseData();
+
+// display posts page 
+function displayItems(dataList, currentPage) {
+    const startIndex = (currentPage - 1) * noOfPosts;
+    const endIndex = startIndex + noOfPosts;
+    const pageItems = dataList.slice(startIndex, endIndex);
+    // console.log(pageItems.length);
+
+    // Clear previous content
+    const container = document.querySelector('.content-container');
+    // if (container.innerHTML!==null ) container.innerHTML='';
+   
+    // Display current page items
+    pageItems.forEach(item => {
+      const itemElement = document.createElement('div');
+      itemElement.setAttribute("class", "blog");
+    //   console.log(item);
+      itemElement.innerHTML = `
+      <h2>${item.title}</h2>
+      <p>${item.body}</p>
+      `
+      container.appendChild(itemElement);
+    }); 
+}
+
+
+
+// create buttons of pages 
+function createPaginatedButton(totalPage){
+    for(let i=1;i<=totalPage;i++){
+        const btn=document.createElement("button");
+        btn.setAttribute("id", `${i}`);
```

Comment:
utils class, handle ( element, creation fn(className, id, style, data)=> element add -> document, return created element, so that incase we want to modify or change anything then we can from the our calling file.

Reply:
Done